src/main/java/com/crud/tasks/trello/client/TrelloClient.java

```diff
@@ -0,0 +1,52 @@
+package com.crud.tasks.trello.client;
+
+import com.crud.tasks.domain.TrelloBoardDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Component
+public class TrelloClient {
+    @Value("${trello.api.endpoint.prod}")
+    private String trelloApiEndpoint;
+
+    @Value("${trello.app.key}")
+    private String trelloAppKey;
+
+    @Value("${trello.app.token}")
+    private String trelloToken;
+
+    @Value("${trello.app.username}")
+    private String trelloUsername;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private String buildUrl(){
+        return trelloApiEndpoint + "/members/" + trelloUsername + "/boards";
+    }
+
+    public List<TrelloBoardDto> getTrelloBoards() {
+        URI url = UriComponentsBuilder.fromHttpUrl(buildUrl())
+                .queryParam("key", trelloAppKey)
+                .queryParam("token", trelloToken)
+                .queryParam("fields", "name,id").build().encode().toUri();
+
+
+        TrelloBoardDto[] boardsResponse = restTemplate.getForObject(url, TrelloBoardDto[].class);
```

Comment:
w przypadku, gdy zapytanie `GET` zwróci jakikolwiek błąd (status z rodziny 4xx, 5xx) to `restTemplate` rzuca wyjątkiem. powinieneś objąć to w `try-catch` i w przypadku błędu zwrócić pustą listę

Reply:
https://www.baeldung.com/spring-rest-template-error-handling